src/autocomplete/AutocompleteView.js

```diff
@@ -34,10 +34,16 @@ export default class AutocompleteView extends PureComponent<Props> {
     const { text, selection } = this.props;
     const result = getAutocompleteFilter(text, selection);
 
-    const AutocompleteComponent = prefixToComponent[result.lastWordPrefix];
+    const { filter, lastWordPrefix } = result;
```

Comment:
I don't like the overuse of destructuring.
One reason we do it for `this.props` is that it saves us two levels of object access (`this` and `props`), and another is that it is simpler and consistent if we use it everywhere.

Reply:
I think in this instance this is helpful; I just pushed 4fe1deae8 which does a version of this, with this explanation why:
```
    autocomplete: Tighten code slightly with destructuring-bind.
    
    Morally this helper `getAutocompleteFilter` is returning two separate
    values, named `lastWordPrefix` and `filter`; the object that collects
    them together doesn't have any real identity of its own, and only
    exists for the purpose of being something we can `return` because JS,
    like most languages, doesn't have a first-class concept of "return
    multiple values".
    
    So, treat it that way at the call site by immediately turning the
    returned object back into the two separate values it represents.
```

Reply:
> Also, when used for props and state only, it is easier to say 'oh, this variable without an object is either local or coming from props', while if using it for everything, that is not the case.

I don't really understand this objection. These are indeed local variables, coming from the return value of a function we called just like local variables often do... what's the notion of "local" that you have in mind that these don't count as?